cypress/integration/CTS/TrialDescriptionGeneralPageComponentsTest/index.js

```diff
@@ -0,0 +1,94 @@
+// <reference types="Cypress" />
+import { Given, And, Then } from 'cypress-cucumber-preprocessor/steps';
+
+And('trial info displays {string}', (infoText) => {
+    const regex = new RegExp(infoText);
+    cy.get('.all-trials').invoke('text').should('match', regex);
+});
+
+And('{string} button does not exist', (showSearchCriteriaButton) => {
+    cy.get('button').contains(showSearchCriteriaButton).should('not.exist');
+});
+
+And('{string} link has a href {string}', (linkText, linkHref) => {
+    cy.get(`a[href="${linkHref}"]`).should('have.text', linkText);
+});
+
+And('{string} button does not exist', (modifySearchCriteriaButton) => {
+    cy.get('button').contains(modifySearchCriteriaButton).should('not.exist');
+});
+
+When('user clicks on {int} result item link', (num) => {
+    cy.get(`div.results-list-item__title a`).eq(num - 1).click();
+});
+
+Then('user is redirected to {string} with query parameters that match {string}', (redirectedPath, queryParam) => {
+    cy.location('pathname').should('eq', redirectedPath);
+    const regex = new RegExp(queryParam);
+    cy.location('search').should('match', regex);
+});
+
+And('{string} link is displayed', (textLink) => {
+    cy.get('div[class*="btnAsLink"]').should('have.text', textLink).and('be.visible');
+});
+
+And('{string} text is displayed', (infoText) => {
+    cy.get('.trial-description-page__header').find('strong').first().should('have.text', infoText).and('be.visible');
+});
+
+And('trial details accordion is displayed', () => {
+    cy.get('.trial-description-page__content  .cts-accordion').should('be.visible');
+});
+
+And('{string} section is expanded', (descriptionSection) => {
+    cy.get('div[class="cts-accordion  "] button.cts-accordion__button').contains(descriptionSection).parent()
```

Comment:
the locator with just ('button.cts-accordion__button').contains... should suffice

Reply:
done